yeoreum/src/components/signUp/Form.tsx

```diff
@@ -0,0 +1,340 @@
+import styled from '@emotion/styled';
+import React, { useCallback, useState } from 'react';
+
+interface AlertProps {
+  success?: boolean;
+}
+
+interface Password {
+  validity: boolean | undefined;
+  message: string;
+}
+
+const Form = () => {
+  const [mailStatus, setMailStatus] = useState(0);
+  const [user, setUser] = useState({
+    email: '',
+    emailCode: '',
+    password: '',
+    passwordConfirm: '',
+  });
+
+  const [emailInfo, setEmailInfo] = useState({
+    validity: true,
+    message: '',
+  });
+  const [emailCodeInfo, setEmailCodeInfo] = useState({
+    validity: false,
+    message: '',
+  });
+  const [passwordInfo, setPasswordInfo] = useState<Password>({
+    validity: undefined,
+    message: '',
+  });
+  const [passwordConfirmInfo, setPasswordConfirmInfo] = useState<Password>({
+    validity: undefined,
+    message: '',
+  });
```

Comment:
- 로직을 다시 한번 고민해봐야 겠지만, 여러 상태로 관리하기보다 같은 관심사를 가진 상태들을 하나의 state로 관리해보면 어떨까싶음! 🙆🏻‍♂️
```suggestion
  const [user, setUser] = useState({
    email: {
      value: '',
      validity: true,
      message: '',
    },
    emailCode: {
      value: '',
      validity: false,
      message: '',
    },
    password: {
      value: '',
      validity: undefined,
      message: '',
    },
    passwordConfirm: {
      value: '',
      validity: undefined,
      message: '',
    },
  });
```